### PR TITLE
fix(frontend react key warnings): add uniq keys to JSX elements

### PR DESCRIPTION
**Pull request by michael-hoffman-26**
### SUMMARY

Add uniq keys to JSX elements,
in order to remove some react warning such as:

image
image

That will improve react performance

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/components/Menu.tsx [hunk 1]
```diff
@@ -237,7 +237,7 @@ export function Menu({
     }
     return (
       <SubMenu
-        key={index}
+        key={`${index} - ${label}`}
```
Comment (etr2460) [inline superset-frontend/src/views/components/Menu.tsx:240]: if these are guaranteed to be unique, let's just use the label here?

Comment (michael-hoffman-26) [reply superset-frontend/src/views/components/Menu.tsx]: Yes good idea, I changed it to `label`, Tnx

**Comment by rusackas** [comment]
Would @eric-briscoe or @kgabryje be able to give this a look, perchance?

**Comment by rusackas** [comment]
@michael-hoffman-26 this will also need a rebase to mitigate code conflicts if there's any intent to merge this.

CC @amitmiran137 for good measure.

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx [hunk 2]
```diff
@@ -132,13 +132,25 @@ const WelcomeNav = styled.div`
...
+      />,
+    );
+  }
+
+  return (
+    <CardContainer showThumbnails={cover} className="loading-cards">
+      {listViewCards}
+    </CardContainer>
+  );
+};
```
Comment (lyndsiWilliams) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:153]:
```suggestion
export const LoadingCards = ({ cover }: LoadingProps) => (
  <CardContainer showThumbnails={cover} className="loading-cards">
    {[...new Array(loadingCardCount)].map((element, index) => (
      <ListViewCard
        key={index}
        cover={cover ? false : <></>}
        description=""
        loading
      />
    ))}
  </CardContainer>
);
```
Would this work the same if you used the iterator in the original `.map`?

**Comment by lyndsiWilliams** [comment]
Closing this PR because all of these are now addressed on the current master branch.

**Pull request closed without merging**